aubio-rs/src/filterbank.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{
+    check_init,
+    Error::{InvalidArg},
+    ffi,
+    vec::{FVecMut, CVec, FMat, FMatVecs},
+    Result,
+    Status,
+};
+
+/**
+ * Filterbank object.
+ *
+ * General-purpose spectral filterbank object.
+ */
+pub struct FilterBank {
+    filterbank: *mut ffi::aubio_filterbank_t,
+    #[cfg(feature = "check-size")]
+    n_filters: usize,
+    #[cfg(feature = "check-size")]
+    win_s: usize,
+}
+
+impl Drop for FilterBank {
+    fn drop(&mut self) {
+        unsafe { ffi::del_aubio_filterbank(self.filterbank) }
+    }
+}
+
+impl FilterBank {
+    /**
+     * Create filterbank object.
+     *
+     * - `n_filters`: Number of filters to create
+     * - `win_s`: size of analysis buffer (and length of the FFT transform)
+     *
+     * Allocates an empty matrix of length win_s / 2 + 1 and height n_filters
+     */
+    pub fn new(n_filters: usize, win_s: usize) -> Result<Self> {
+        let filterbank = unsafe {
+            ffi::new_aubio_filterbank(n_filters as ffi::uint_t, win_s as ffi::uint_t)
+        };
+
+        check_init(filterbank)?;
+
+        #[cfg(feature = "check-size")]
+        return Ok(Self { filterbank, n_filters, win_s });
```

Comment:
To avoid `return` keyword here you can enclose both expressions with `{}`

```rust
#[cfg(feature = "check-size")]
{
    Ok(Self { filterbank, n_filters, win_s })
}

#[cfg(not(feature = "check-size"))]
{
    Ok(Self { filterbank })
}
```